projects/kit/src/lib/processors/normalize-date-preprocessor.ts

```diff
@@ -14,53 +14,52 @@ export function normalizeDatePreprocessor({
     dateTimeSeparator?: string;
 }): MaskitoPreprocessor {
     return ({elementState, data}) => {
-        const separator = rangeSeparator
-            ? new RegExp(`${rangeSeparator}|-`)
-            : dateTimeSeparator;
-        const possibleDates = data.split(separator);
-        const dates = data.includes(dateTimeSeparator)
-            ? [possibleDates[0] ?? '']
-            : possibleDates;
+        const dateSegments = data.split(/\D/).filter(Boolean);
+        const templateSegments = dateModeTemplate.split(dateSegmentsSeparator);
+        const includesTime = data.includes(dateTimeSeparator);
 
-        if (
-            dates.every(
-                (date) =>
-                    date.trim().split(/\D/).filter(Boolean).length ===
-                    dateModeTemplate.split(dateSegmentsSeparator).length,
-            )
-        ) {
-            const newData = dates
-                .map((date) =>
-                    normalizeDateString(date, dateModeTemplate, dateSegmentsSeparator),
-                )
-                .join(rangeSeparator);
+        let newData = data;
```

Comment:
Do we actually need to create it here?

Could we just write in following way ?
```ts
if (firstCodition) {
    return {elementState, date: anotherCondition ? 'option1' : 'option2'};
}

return {elementState, data};
```